### PR TITLE
AG-8492 Add examples typescript check to build process

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-8492

The main changes are https://github.com/ag-grid/ag-charts/commit/9f820f2574f1ae16b4158bbe631ca09648015a5b & https://github.com/ag-grid/ag-charts/commit/b42868e7fb1f84a88ea9046d004bda470f8b0a05. The rest are fixes these caught.

### CODE REVIEW

File: packages/ag-charts-website/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "extends": "../../tsconfig.base.json",
+  "include": ["./src/content/docs/**/_examples/**/*.ts", "./src/content/gallery/_examples/**/*.ts"],
```
Comment (lsjroberts) [inline packages/ag-charts-website/tsconfig.json:3]: This excludes the rest of the website package, however there are currently 800 ts errors in that. But none of those are used in the library so not a big deal.

This does though hinder some behaviour in vscode around our usage of `import foo from @design-system/foo` etc. Not a major issue and we can fix up all the ts errors in the website at some point later.

File: packages/ag-charts-website/project.json [hunk 2]
```diff
@@ -17,14 +17,9 @@
...
-      "inputs": ["tsDefaults"],
-      "outputs": ["{options.outputPath}"],
+      "executor": "nx:run-commands",
       "options": {
-        "main": "packages/ag-charts-website/src",
-        "outputPath": "dist/packages/ag-charts-website/out-tsc",
-        "tsConfig": "packages/ag-charts-website/tsconfig.json",
-        "compiler": "tsc"
+        "command": "tsc --project packages/ag-charts-website/tsconfig.json"
       }
```
Comment (alantreadway) [inline packages/ag-charts-website/project.json:23]: What was the motivation to switch to the `tsc` command-line tool? I'd like to try and minimise this scenario, as it adds more edge cases on upgrade vs. following the 'standard' Nx approach?

Comment (lsjroberts) [reply packages/ag-charts-website/project.json:23]: I'm not sure what the nx/js:tsc is running, but it doesn't output any of the errors and passes while `tsc` fails correctly.

Comment (alantreadway) [reply packages/ag-charts-website/project.json:23]: We bounced around a bit on this offline, and are unable to find a way to use the Nx approach to resolve this issue - lets move ahead with the proposed change :+1:

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**